### PR TITLE
Upgrade dependencies and overhaul project configuration structure

### PR DESCRIPTION
**Pull request by tbcrawford**
Goals of this PR:
- Upgrade core project dependency: embedded-kafka to latest version
- Upgrade Kotlin from 1.4.31 to 1.6.21 (this matches kotest core)
- Upgrade dependency management configuration to use version catalogs (this should help renovate unless I can't see what renovate has been trying to do, but it doesn't look like it's been suggesting anything)
- Switch to latest gradle build practices (version catalog, settings.gradle.kts, convention plugins)
- Continued support of jvm target 1.8
- Upgrade project gradle version from 7.5.1 to 7.6
- Fix code using deprecated code

Given the major upgrade of embedded-kafka I propose that this extension should go to v2.0.0, but v1.1.0 might be suitable as well (I'll leave that up to the maintainers).

### CODE REVIEW

File: buildSrc/src/main/kotlin/Ci.kt [hunk 1]
```diff
@@ -1,8 +1,7 @@
 object Ci {
-
    // this is the version used for building snapshots
    // .GITHUB_RUN_NUMBER-snapshot will be appended
-   private const val snapshotBase = "1.0.0"
+   private const val snapshotBase = "2.0.0"
```
Comment (tbcrawford) [inline buildSrc/src/main/kotlin/Ci.kt:4]: Just wanted to make a call out that I changed this to version 2.0.0 here. Please let me know if this should be reverted or changed to something else. I'd be happy to make the change.

**Review by jonasg13 — APPROVED** [review]

File: buildSrc/src/main/kotlin/kotest-publishing-conventions.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+java {
+   withJavadocJar()
+   withSourcesJar()
+}
+
+publishing {
+   repositories {
+      maven {
+         val releasesRepoUrl = uri("https://oss.sonatype.org/service/local/staging/deploy/maven2/")
+         val snapshotsRepoUrl = uri("https://oss.sonatype.org/content/repositories/snapshots/")
```
Comment (Kantis) [inline buildSrc/src/main/kotlin/kotest-publishing-conventions.gradle.kts]:
```suggestion
         val releasesRepoUrl = uri("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
         val snapshotsRepoUrl = uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
```

Sonatype changed the URL for publishing

File: buildSrc/src/main/kotlin/Ci.kt (hunk 1 above)
Comment (Kantis) [reply buildSrc/src/main/kotlin/Ci.kt:4]: I think it's a good idea to go for 2.0 👍🏼

File: settings.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+rootProject.name = "kotest-extensions-embedded-kafka"
+
+dependencyResolutionManagement {
+   repositories {
+      mavenLocal()
+      mavenCentral()
+      maven {
+         url = uri("https://oss.sonatype.org/content/repositories/snapshots")
```
Comment (Kantis) [inline settings.gradle.kts]:
```suggestion
         url = uri("https://s01.oss.sonatype.org/content/repositories/snapshots")
```

File: settings.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      }
+   }
+}
+
+pluginManagement {
+   repositories {
+      gradlePluginPortal()
+      mavenCentral()
+      maven {
+         url = uri("https://oss.sonatype.org/content/repositories/snapshots")
```
Comment (Kantis) [inline settings.gradle.kts]:
```suggestion
         url = uri("https://s01.oss.sonatype.org/content/repositories/snapshots")
```

File: buildSrc/settings.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+rootProject.name = "buildSrc"
+
+dependencyResolutionManagement {
+   repositories {
+      mavenLocal()
+      mavenCentral()
+      maven {
+         url = uri("https://oss.sonatype.org/content/repositories/snapshots")
```
Comment (Kantis) [inline buildSrc/settings.gradle.kts]:
```suggestion
         url = uri("https://s01.oss.sonatype.org/content/repositories/snapshots")
```

File: buildSrc/settings.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      }
+   }
+}
+
+pluginManagement {
+   repositories {
+      gradlePluginPortal()
+      mavenCentral()
+      maven {
+         url = uri("https://oss.sonatype.org/content/repositories/snapshots")
```
Comment (Kantis) [inline buildSrc/settings.gradle.kts]:
```suggestion
         url = uri("https://s01.oss.sonatype.org/content/repositories/snapshots")
```

**Review by Kantis — APPROVED** [review]
Great contribution, thanks! 🚀

File: buildSrc/src/main/kotlin/Ci.kt (hunk 1 above)
Comment (tbcrawford) [reply buildSrc/src/main/kotlin/Ci.kt:4]: Excellent! I'll leave that as it is. I updated the sonatype url. Do you need me to update the README as well to mention something about v2? Is there anything else you need before we could release v2 here?

Comment (Kantis) [reply buildSrc/src/main/kotlin/Ci.kt:4]: Updating readme would be appreciated. I think I can take care of the rest. 😃

Comment (tbcrawford) [reply buildSrc/src/main/kotlin/Ci.kt:4]: Excellent! I have updated the README to include 2.0.0 release notes 🙂

**Pull request merged**